### DISCUSSION TITLE
does icu4x support selectordinal?

### DISCUSSION THREAD

**Discussion by ghost** [unicode-org/icu4x#2806] [Q&A] [answered]
no related code snippets or issue found after searching.

**Comment by zbraniecki** [comment, marked as answer]
Hi @byte1234 ! Thank you for interest in ICU4X.

Yes, we support it via [`icu::plurals::PluralRules`](https://icu4x.unicode.org/doc/icu/plurals/struct.PluralRules.html).

In the snippet example it uses [`PluralRuleType::Cardinal`](https://icu4x.unicode.org/doc/icu/plurals/enum.PluralRuleType.html) which you can swap for `Ordinal` to get that instead.

Compared to ICU4C/J and ECMA-402 we renamed the method `select` to [`category_for`](https://icu4x.unicode.org/doc/icu/plurals/struct.PluralRules.html#method.category_for) to leave space for deeper future integration into upcoming MessageFormat 2 plural selection model.

I apologize for the inconvenience and hope this helps. Please let me know if it resolved your issue

**Reply by ghost** [reply]
wow, really thanks. Btw, how does icu4x parse the messageformat, I find
that the icu4c contains the code for that. If it isn’t a builtin feature,
may i ask for a recommendation? Thank you, We are considering icu4x for our
cross platform i18n solution.

**Reply by zbraniecki** [reply]
We do not have support for MessageFormat yet. Unicode is working on a revision of MF called [MessageFormat 2](https://github.com/unicode-org/message-format-wg) which is a generational advancement over the previous revision and ICU4X's plan is to directly support the new revision.

I started [prototyping it](https://github.com/unicode-org/icu4x/pull/2272), but had to pause to prioritize a more urgent feature (generic pattern support). I plan to resume work on MF2 in ICU4X in a month or so.
